src/docs/release/breaking-changes/use-maxLengthEnforcement-instead-of-maxLengthEnforced.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Use maxLengthEnforcement instead of maxLengthEnforced
+description: Introduing MaxLengthEnforcement enum.
+---
+
+## Summary
+
+_If you didn't use `maxLengthEnforced` in `TextField`/`TextFormField`/`CupertinoTextField`, or you don't care about the truncate behavior for the composing text editing value, you can stop reading._
+
+To control the behavior of `maxLength` in the `LengthLimitingTextInputFormatter`, use `maxLengthEnforcement` instead of `maxLengthEnforced`, and `maxLengthEnforced` is deprecated.
+
+## Context
+
+Before [the first PR][Fix TextField crashed with composing and maxLength set], `LengthLimitingTextInputFormatter` will always truncate the editing value, which may leads to engine crashes when the composing range is overflowed sometimes. With this changes, a composing `TextEditingValue` will not be truncated if only `maxLength` set, and still be truncated after the composing ends (See [`maxLength` documentation][] for more specific behavior). But it's not optional.
```

Comment:
I don't think the square brackets work the way you think. The text in the second set of square brackets is not displayed. It serves only as a shortcut. So:

```
Before [PR 63754][]: Fix TextField crashed with composing and maxLength set,
the `LengthLimitingTextInputFormatter` always truncated the editing value,
which resulted in engine crashes when the composing range sometimes overflowed.
With this change, an editable `TextEditingValue` is not truncated if only `maxLength` is set,
but _is_ truncated after editing is complete. (For more specific behavior, see the
[`maxLength` documentation][]. 

[PR 63754]:  {{site.github.com}}//flutter/flutter/pull/63754
```

I deleted the "But it's not optional" line because I don't understand. What isn't optional?

Reply:
> I deleted the "But it's not optional" line because I don't understand. What isn't optional?

@sfshaza2 The behavior, allow composing exempt, is not optional. It provides as `allowComposingTextToFinish` enforcement now.